src/uipath/_cli/_evals/_progress_reporter.py

```diff
@@ -344,6 +505,55 @@ def _collect_results(
             )
         return assertion_runs, evaluator_scores_list
 
+    def _collect_coded_results(
```

Comment:
Let's avoid duplication of code? High overlap between this and `_collect_results`.